### PR TITLE
[SPARK-43151][DOC] Update the prerequisites for generating Python API docs

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Update the prerequisites for generating Python API docs:
* The command should be run under the docs directory so that the input file should be `../dev/requirements.txt`
* Remove `sudo` in the command
* Remove the version for `torch` and `torchvision`. I was getting 

```
ERROR: Could not find a version that satisfies the requirement torch==1.13.1 (from versions: 2.0.0)
ERROR: No matching distribution found for torch==1.13.1
ERROR: Could not find a version that satisfies the requirement torchvision==0.14.1 
```

### Why are the changes needed?

Update the prerequisites for generating Python API docs, to save troubles from other developers when generating Python Docs.

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

Manually try the command and new requirements.txt

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ See also https://issues.apache.org/jira/browse/SPARK-35375.
 -->
 Run the following command from $SPARK_HOME:
 ```sh
-$ sudo pip install --upgrade -r dev/requirements.txt
+$ pip install --upgrade -r ../dev/requirements.txt
```
Comment (amaliujia) [inline docs/README.md:64]: It syas `Run the following command from $SPARK_HOME:` so I think `dev/requirements.txt` is a proper path.

Comment (gengliangwang) [reply docs/README.md]: ok, updated

**Comment by amaliujia** [comment]
LGTM

**Comment by zhengruifeng** [comment]
@gengliangwang I just success to run this `pip install --upgrade -r dev/requirements.txt` in a clean conda env, and can not reproduce the issue related to pytorch version.

@HyukjinKwon @WeichenXu123 have you seen this version issue?

**Comment by HyukjinKwon** [comment]
I haven't seen this before. but I think loosing the version restrictions looks fine.

File: dev/requirements.txt [hunk 2]
```diff
@@ -61,5 +61,5 @@ googleapis-common-protos-stubs==2.2.0
 grpc-stubs==1.24.11
 
 # TorchDistributor dependencies
-torch==1.13.1
-torchvision==0.14.1
+torch
+torchvision
```
Comment (HyukjinKwon) [inline dev/requirements.txt:65]: @rithwik-db fyo

**Review by HyukjinKwon — APPROVED** [review]

**Comment by gengliangwang** [comment]
> @gengliangwang I just success to run this pip install --upgrade -r dev/requirements.txt in a clean conda env, and can not reproduce the issue related to pytorch version.

@zhengruifeng I am using python 3.11.2
And I am actually running with 
```
pip3 install --upgrade -r dev/requirements.txt
```

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**